Packs/GoogleChronicleBackstory/Integrations/GoogleChronicleBackstory/GoogleChronicleBackstory.py

```diff
@@ -1108,6 +1135,916 @@ def get_list_events_hr(events):
+
+    return events_ec
+
+
+def get_context_for_detections(detection_resp: Dict[str, Any]) -> Tuple[List[Dict[str, Any]], Dict[str, str]]:
+    """
+    Convert detections response into Context data
+
+    :param detection_resp: Response fetched from the API call for detections
+    :type detection_resp: Dict[str, Any]
+
+    :return: list of detections and token to populate context data
+    :rtype: Tuple[List[Dict[str, Any]], Dict[str, str]]
+    """
+    detections_ec = []
+    token_ec = {}
+    next_page_token = detection_resp.get('nextPageToken')
+    if next_page_token:
+        token_ec = {'name': 'gcb-list-detections', 'nextPageToken': next_page_token}
+    detections = detection_resp.get('detections', [])
+    for detection in detections:
+        detection_dict = detection
+        result_events = detection.get('collectionElements', [])
+        if result_events:
+            detection_dict['collectionElements'] = get_events_context_for_detections(result_events)
+
+        detection_details = detection.get('detection', {})
+        if detection_details:
+            detection_dict.update(detection_details[0])
+            detection_dict.pop('detection')
+
+        time_window_details = detection.get('timeWindow', {})
+        if time_window_details:
+            detection_dict.update({'timeWindowStartTime': time_window_details.get('startTime'),
+                                   'timeWindowEndTime': time_window_details.get('endTime')})
+            detection_dict.pop('timeWindow')
+        detections_ec.append(detection_dict)
+
+    return detections_ec, token_ec
+
+
+def get_detections(client_obj, rule_or_version_id: str, page_size: str, detection_start_time: str,
+                   detection_end_time: str, page_token: str, alert_state: str) \
+        -> Tuple[Dict[str, Any], Dict[str, Any]]:
+    """
+    Returns context data and raw response for gcb-list-detections command.
+
+    :type client_obj: Client
+    :param client_obj: client object which is used to get response from api
+    :type rule_or_version_id: str
+    :param rule_or_version_id: rule_id or version_id to fetch the detections for.
+    :type page_size: str
+    :param page_size: Number of detections to fetch at a time.
+    :type detection_start_time: str
+    :param detection_start_time: The time to start listing detections from.
+    :type detection_end_time: str
+    :param detection_end_time: The time to start listing detections to.
+    :type page_token: str
+    :param page_token: The token for the page from which the detections should be fetched.
+    :type alert_state: str
+    :param alert_state: Alert state for the detections to fetch.
+
+    :rtype: Tuple[Dict[str, Any], Dict[str, Any]]
+    :return: ec, json_data: Context data and raw response for the fetched detections
+    """
+    # Make a request URL
+    request_url = '{}/detect/rules/{}/detections?pageSize={}' \
+        .format(BACKSTORY_API_V2_URL, rule_or_version_id, page_size)
```

Comment:
I'm not sure about this use of the V2 URL. Could you explain it please?

Reply:
@guykeller For fetching the detections, we have the endpoint corresponding to version 2 of Chronicle's API. So, for any future implementations for this API version, we have added it as a constant here instead of hardcoding it while making the API call for fetching detections.

Reply:
> I'm not sure about this use of the V2 URL. Could you explain it please?

@guykeller For fetching the detections, we have the endpoint corresponding to version 2 of Chronicle's API. So, for any future implementations for this API version, we have added it as a constant here instead of hardcoding it while making the API call for fetching detections.